### ISSUE TITLE
It seams that cache does not help Improve query efficiency

### ISSUE THREAD

**Issue by HwiLu** [apache/druid#13006]
**Druid Version:  apache-druid-0.23.0  single-server/xlarge**
**historical heapsize: 256g**

We were doing choosing database for  our system, and we did a performance test on druid, and found it seams that cache does not help Improve query efficiency.
We submit multiple queries concurrently to druid with script like below
```sh
#!/bin/bash
echo "input ts qs"
ts=$1
qs=$2
QUEUE=`cat /opt/druid/luhw/table.txt.$ts`
THREAD=$ts
TIMEFILE=multi_druid_query.time${ts}_${qs}
>$TIMEFILE
rm -f ./druidResult/*


TMPFIFO=/tmp/$$.fifo			         
mkfifo $TMPFIFO                
exec 5<>${TMPFIFO}             
rm -rf ${TMPFIFO}             

for((i=1;i<=$THREAD;i++))
do
   echo ;
done >&5 

for sqlFile in $QUEUE          
do
        read -u5        
        {			
        cat >./json/${sqlFile}_query.json <<EOF
{"query": "select TradeTime AS ts,TradeTime as exch_ts,TradeVolume as sz,SAFE_DIVIDE(TradeAmount,TradeVolume)/10 as px from OPT_XDCE WHERE \"__time\" > '2022-08-08T00:00:00.000Z' and \"__time\" < '2022-08-08T23:59:59.999Z'  and idea_instrument='$sqlFile' order by __time asc"}
EOF

        (time curl -X 'POST' -H 'Content-Type:application/json' -d @/opt/druid/luhw/json/${sqlFile}_query.json http://admin:admin@192.168.1xx.xx:8888/druid/v2/sql 1>> druidResult/test_all_result.json)  2>> $TIMEFILE
          
        echo "" >&5    
        } &
done
wait               
exec 5>&-          
exit 0
```
the core statement is `select TradeTime AS ts,TradeTime as exch_ts,TradeVolume as sz,SAFE_DIVIDE(TradeAmount,TradeVolume)/10 as px from OPT_XDCE WHERE \"__time\" > '2022-08-08T00:00:00.000Z' and \"__time\" < '2022-08-08T23:59:59.999Z'  and idea_instrument='$sqlFile' order by __time asc`

and then I enable pre-cache and whole-cache one by one, and calculate average and maximum time of the queries, I found enable  and disable the cache efficiency is not much different, just like the result as follows

![image](https://user-images.githubusercontent.com/25824870/187639340-4b185c69-1b20-42a4-a0c3-32f58194c3e4.png)

Does anyone have any idea on this case? and is there anything wrong with the tests I did?

Best wishes.

**Comment by abhishekagarwal87** [comment]
There is no caching for scan queries. I didn't know it beforehand but as I think about it, it makes sense to not cache the results for scan queries. For the following reasons
- Results for scan queries are streamed back to the client. Caching the results of the scan query is going to put memory pressure on the broker. 
- There is no benefit of caching scan queries. For a group-by query, you could be potentially reading millions of rows and packing them into a few rows. The advantage of caching can be significant, especially at the segment level. (https://druid.apache.org/docs/0.23.0/querying/caching.html). With scan, it's unlikely that the user is going to scan millions of rows back to the client. Now one can say that we could potentially save the time spent on filtering. But that cost is cheap enough because of the bitmap indices that segments have. 

what is the target number you are shooting for? are most of your queries going to scan queries or group-by queries? If the latter, you can re-run the same tests with group-by queries and see the difference. There is also some work done recently to improve QPS that is going to be available in the 24.0.0 release. One such change is - https://github.com/apache/druid/pull/12514

**Comment by gianm** [comment]
I wouldn't say there is _no_ benefit to caching scan queries. But certainly the expected benefit (vs. cost) is higher for queries that do aggregations. I expect we'll add caching for scan queries at some point, in a way that mitigates the costs, like by having a max payload size that we'll cache.

Btw, based on reading your query, the number one, most effective thing you can do to get good performance is secondary partitioning on `idea_instrument`. See https://imply.io/blog/multi-dimensional-range-partitioning/ for details on why this is great. If you are not already doing this I strongly recommend it. In current releases you can do this with `partitionsSpec` on batch ingest like:

```json
"partitionsSpec" : {
  "type": "range",
  "partitionDimensions":["idea_instrument"],
  "targetRowsPerSegment" : 5000000
}
```

In next release (24.0) you can do this with SQL-based ingest using:

```
CLUSTERED BY idea_instrument
```
